tests/test_cssselect.py

```diff
@@ -453,6 +468,16 @@ def xpath(css):
         assert xpath("e:EmPTY") == ("e[not(*) and not(string-length())]")
         assert xpath("e:root") == ("e[not(parent::*)]")
         assert xpath("e:hover") == ("e[0]")  # never matches
+        assert (
+            xpath("div:has(div.foo)") == "div[descendant::div]"
+            "[@class and contains(concat(' ', normalize-space(@class), ' '), ' foo ')]"
+        )
```

Comment:
Is this correct? Shouldn't it be `div[descendant::div[@class and contains(concat(' ', normalize-space(@class), ' '), ' foo ')]]` as the class condition is for the descendant?